app/models/concerns/create_parent_object.rb

```diff
@@ -116,7 +117,9 @@ def create_new_parent_csv
       rescue StandardError => e
         batch_processing_event("Skipping row [#{index + 2}] Unable to save parent: #{e.message}.", "Skipped Row")
       end
+      index + 1 if index + 1 - start_index > BatchProcess::BATCH_LIMIT
```

Comment:
You need a return on these to break out of the loop. Otherwise it will keep going until it gets to the end of the list.